lldb/source/Core/ValueObject.cpp

```diff
@@ -414,13 +417,17 @@ lldb::ValueObjectSP ValueObject::GetChildAtIndexPath(
   if (idxs.size() == 0)
     return GetSP();
   ValueObjectSP root(GetSP());
+
+  size_t current_index = 0;
   for (std::pair<size_t, bool> idx : idxs) {
     root = root->GetChildAtIndex(idx.first, idx.second);
     if (!root) {
       if (index_of_error)
-        *index_of_error = idx.first;
+        *index_of_error = current_index;
       return root;
     }
+
+    current_index += 1;
```

Comment:
I'm actually going to go with `++current_index` because that's what [the LLVM coding style](https://llvm.org/docs/CodingStandards.html#prefer-preincrement) recommends.


> Hard fast rule: Preincrement (++X) may be no slower than postincrement (X++) and could very well be a lot faster than it. Use preincrementation whenever possible.
>
> ...

Reply:
Note this guideline is about pre vs post, and it refers to iterators mostly.

I don't think you need to change anything here!

Reply:
Right, but if we're not going with `+= 1` (for consistency and/or informal style?), then I prefer to use the pre-increment operator.

Or are you saying we should just stick with `+= 1` now?

Reply:
Ok, I'll just stick with `+= 1` and hope it becomes a Swift-inspired trend.